ledger/participant-state/kvutils/tools/entries-extractor/src/replay/scala/ledger/participant/state/kvutils/tools/extractor/Extractor.scala

```diff
@@ -0,0 +1,126 @@
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.participant.state.kvutils.tools.extractor
+
+import com.daml.ledger.participant.state.kvutils.Conversions._
+import com.daml.ledger.participant.state.kvutils.export.{
+  ProtobufBasedLedgerDataImporter,
+  SubmissionInfo,
+}
+import com.daml.ledger.participant.state.kvutils.wire.DamlSubmission
+import com.daml.ledger.participant.state.kvutils.{Envelope, Raw}
+import com.daml.lf.data._
+import com.daml.lf.testing.snapshot.Snapshot
+
+import java.io.BufferedOutputStream
+import java.nio.file.{Files, Path, Paths}
+import scala.jdk.CollectionConverters._
+import scala.util.control.NonFatal
+
+object Extractor extends App {
```

Comment:
Do you plan to add some tests for it?

Reply:
Unfortunate I cannot test it easily in the main repo, as there is not more ledger export generated here.
Once you move the tool to the integration kit, I will be please to write tests there. 

Note I have tested manually.